poc/mastic.py

```diff
@@ -277,35 +271,21 @@ def prep_init(
                     vec_sub(n.w, vec_add(n.left_child.w, n.right_child.w)))
                 q += [n.left_child, n.right_child]
 
-            # Update the onehot check, consisting of the hash of the
-            # node proofs in breadth-first order. Each update
-            # resembles a step of Merkle-Damgard compression. The main
-            # difference is that we XOR each block (i.e., corrected
-            # node proof) with the previous hash (or IV) rather than
-            # compress.
-            #
-            #                node proof
-            #                 |
-            #                 v
-            # current     +-----+     +------+     +-----+    updated
-            # proof --+-->| XOR |---->| Hash |---->| XOR |--> proof
-            #         |   +-----+     +------+     +-----+
-            #         |                               ^
-            #         |                               |
-            #         +-------------------------------+
-            #
-            # [MST24]: \tilde\pi = H_1(x^{\leq i} || s^\i)
-            #          \pi = \tilde \pi \xor
-            #             H_2(\pi \xor (\tilde\pi \xor t^\i \cdot \cs^\i)
-            onehot_proof = xor(
-                onehot_proof, self.hash_proof(ctx, xor(onehot_proof, n.proof)))
+            # Update the onehot check.
+            onehot_check_binder += n.proof
 
         payload_check = self.xof(
             b'',
             dst_alg(ctx, USAGE_PAYLOAD_CHECK, self.ID),
             payload_check_binder,
         ).next(PROOF_SIZE)
 
+        onehot_check = self.xof(
```

Comment:
Are we calling it `check` from now on? Leaving a note to remember to update the .md

Reply:
Open for debate. I prefer "check" for consistency with "payload check" and "counter check".

Reply:
fine by me.